UI/js-src/lsmb/DateTextBox.js

```diff
@@ -16,15 +17,37 @@ define([
             if (! params.constraints) {
                params.constraints = { };
             }
-            if (! params.constraints.datePattern) {
+            if (! params.constraints.datePattern
+                 && lsmbConfig.dateformat) {
               params.constraints.datePattern =
                 lsmbConfig.dateformat.replace(/mm/,"MM");
             }
-            if (! params.placeholder) {
+            if (! params.placeholder && lsmbConfig.dateformat) {
                params.placeholder = lsmbConfig.dateformat;
             }
             // end of 'old code' support block
 
+            // retrieve format to add it as the placeholder
+            // (unless there's a placeholder already)
+            if (! params.placeholder) {
+               var l = i18n.normalizeLocale(params.locale),
+                   formatLength = params.formatLength || 'short',
+                   bundle = locale._getGregorianBundle(l);
+
+               if(params.constraints.selector == "year") {
+                  params.placeholder = bundle["dateFormatItem-yyyy"] || "yyyy";
+               }
+               else if(params.constraints.selector == "time") {
+                  params.placeholder = params.constraints.timePattern
+                       || bundle["timeFormat-"+formatLength];
+               }
+               else {
+                  params.placeholder = params.constraints.datePattern
+                       || bundle["dateFormat-"+formatLength];
+               }
+               params.placeholder =
+                  params.placeholder.replace(/M/g, 'm').replace(/y/g, 'yy');
+            }
             this.inherited(arguments);
```

Comment:
As per documentation, constructors must not call inherited. See https://dojotoolkit.org/reference-guide/1.8/dojo/_base/declare.html#id8